File: problems/[E]268.Missing-Number.kt
/**
 * Runtime: 196 ms, faster than 98.69% of Kotlin online submissions for Missing Number.
 * Memory Usage: 37 MB, less than 85.62% of Kotlin online submissions for Missing Number.
 */
class Solution {
    fun missingNumber(nums: IntArray): Int {
        val numbersCount = nums.size
        var idealSum = 0
        for (i in 1 .. numbersCount) {
            idealSum += i
        }
        var sum = 0
        for (i in 0 until numbersCount) {
            sum += nums[i]
        }
        
        return idealSum - sum
    }
}

File: swift/[E]268.Missing-Number.swift
/**
 * Runtime: 124 ms, faster than 100.00% of Swift online submissions for Missing Number.
 * Memory Usage: 14.2 MB, less than 75.99% of Swift online submissions for Missing Number.
 */
class Solution {
    func missingNumber(_ nums: [Int]) -> Int {
        var expectedSum = nums.count
        var totalSum = 0
        for i in 0...(nums.count - 1) {
            expectedSum += i
            totalSum += nums[i]
        }
        return expectedSum - totalSum
    }
}
